x-pack/plugins/infra/public/components/logging/log_text_stream/loading_item_view.tsx

```diff
@@ -28,122 +44,202 @@ export class LogTextStreamLoadingItemView extends React.PureComponent<
+          />
+        ) : null}
+      </EuiFlexGroup>
+    );
+
+    return (
+      <ProgressEntryWrapper className={className} position={position}>
+        {position === 'start' ? <>{extra}</> : null}
+        <ProgressMessage timestamp={timestamp} />
+        {position === 'end' ? <>{extra}</> : null}
+      </ProgressEntryWrapper>
+    );
   }
 }
 
-interface ProgressEntryProps {
-  alignment: 'top' | 'bottom';
-  className?: string;
-  color: 'subdued' | 'primary';
-  isLoading: boolean;
-}
+const ProgressEntryWrapper = euiStyled.div<{ position: Position }>`
+  padding-left: ${props => props.theme.eui.euiSizeS};
+  padding-top: ${props =>
+    props.position === 'start' ? props.theme.eui.euiSizeL : props.theme.eui.euiSizeM};
+  padding-bottom: ${props =>
+    props.position === 'end' ? props.theme.eui.euiSizeL : props.theme.eui.euiSizeM};
+`;
+
+const ProgressMessage: React.FC<{ timestamp?: number }> = ({ timestamp }) => {
+  return (
+    <LogTextSeparator>
+      <EuiTitle size="xxs">
+        {timestamp ? (
+          <FormattedMessage
+            id="xpack.infra.logs.showingEntriesUntilTimestamp"
+            defaultMessage="Showing entries until {timestamp}"
+            values={{ timestamp: <FormattedTime value={timestamp} /> }}
+          />
+        ) : (
+          <FormattedMessage
+            id="xpack.infra.logs.noAdditionalEntriesFoundText"
+            defaultMessage="No additional entries found"
+          />
+        )}
+      </EuiTitle>
+    </LogTextSeparator>
+  );
+};
+
+const ProgressSpinner: React.FC<{ kind: 'streaming' | 'loading' }> = ({ kind }) => (
+  <>
+    <EuiFlexItem grow={false}>
+      <EuiLoadingSpinner size="l" />
+    </EuiFlexItem>
+    <EuiFlexItem grow={false}>
+      <EuiText size="s">
+        {kind === 'streaming' ? (
+          <FormattedMessage
+            id="xpack.infra.logs.streamingNewEntriesText"
+            defaultMessage="Streaming new entries"
+          />
+        ) : (
+          <FormattedMessage
+            id="xpack.infra.logs.loadingNewEntriesText"
+            defaultMessage="Loading new entries"
+          />
+        )}
+      </EuiText>
+    </EuiFlexItem>
+  </>
+);
+
+const ProgressCta: React.FC<Pick<
+  LogTextStreamLoadingItemViewProps,
+  'position' | 'rangeEdge' | 'onExtendRange' | 'onStreamStart'
+>> = ({ position, rangeEdge, onExtendRange, onStreamStart }) => {
+  if (rangeEdge === 'now' && position === 'end') {
+    return (
+      <EuiButton onClick={onStreamStart} size="s">
+        <FormattedMessage id="xpack.infra.logs.streamLive" defaultMessage="Stream live" />
+      </EuiButton>
+    );
+  }
+
+  const iconType = position === 'start' ? 'arrowUp' : 'arrowDown';
+
+  if (!rangeEdge) {
+    return null;
+  }
 
-const ProgressEntry: React.FC<ProgressEntryProps> = props => {
-  const { alignment, children, className, color, isLoading } = props;
+  const extendedRange = extendDatemath(rangeEdge, position === 'start' ? 'before' : 'after');
```

Comment:
This seems to be `undefined` in some situations when I would have expected it to show up:

![grafik](https://user-images.githubusercontent.com/973741/75374949-77d1ce80-58cd-11ea-965a-176c39f20a27.png)

Here with `rangeEdge === '"2020-02-26T17:40:00.000Z"'` and `position === 'start'.

Reply:
I left it out on purpose for absolute units. I couldn't come up with a good answer to "how much should it extend?"

Some ideas I came up with:

- Make it dynamic based on the distance from beginning to end. This is tricky because the other edge can be relative.
- Round to the next pretty number. For example if the user enters `... .347`, round to the previous second. If the second is `:47.000` round to the previous minute, etc. It gets tricky though if they enter a nice round number, like midnight.

Reply:
> Make it dynamic based on the distance from beginning to end. This is tricky because the other edge can be relative.

I think it's reasonable to use the interval size at the moment of rendering. In the acceptance criteria of the implementation issue I expressed it like

> extend the time frame into their respective direction based on the current time frame size (e.g. if the selected timeframe is <= 1 day, extend by 1 hour. If it is <= 1 week extend by 1 day, etc).

Reply:
> Make it dynamic based on the distance from beginning to end. This is tricky because the other edge can be relative.

I think it's reasonable to use the interval size at the moment of rendering. In the acceptance criteria of the implementation issue I expressed it like

> extend the time frame into their respective direction based on the current time frame size (e.g. if the selected timeframe is <= 1 day, extend by 1 hour. If it is <= 1 week extend by 1 day, etc).